lambdas/pkgselect/index.py

```diff
@@ -56,10 +56,16 @@ def file_list_to_folder(df: pd.DataFrame) -> dict:
     lambda).
     """
     try:
-        folder = pd.Series(df.logical_key.dropna().str.extract('([^/]+/?).*')[0].unique())
-        prefixes = folder[folder.str.endswith('/')].sort_values().tolist()
-        objects = folder[~folder.str.endswith('/')].sort_values().tolist()
-    except AttributeError:
+        groups = df.groupby(df.logical_key.str.extract('([^/]+/?).*')[0], dropna=True)
+        folder = groups.agg(
+            size=('size', 'sum'),
+            physical_key=('physical_key', 'first')
+        )
+        folder.reset_index(inplace=True)
+        folder.rename(columns={0: 'logical_key'}, inplace=True)
+        prefixes = folder[folder.logical_key.str.contains('/')].to_dict(orient='records')
+        objects = folder[~folder.logical_key.str.contains('/')].to_dict(orient='records')
```

Comment:
What happens in keys with successive `//` to both the regex in the groupby and here?

Reply:
// is illegal in logical keys isn't it?